packages/cli/src/dbt/profile.ts

```diff
@@ -0,0 +1,49 @@
+import { CreateWarehouseCredentials, ParseError } from 'common';
+import { readFileSync } from 'fs';
+import * as yaml from 'js-yaml';
+import * as path from 'path';
+import { convertPostgresSchema } from './targets/postgres';
+import { convertSnowflakeSchema } from './targets/snowflake';
+import { renderProfilesYml } from './templating';
+import { LoadProfileArgs, Profiles, Target } from './types';
+
+export const loadDbtTarget = ({
+    profilesDir,
+    profileName,
+    targetName,
+}: LoadProfileArgs): Target => {
+    const profilePath = path.join(profilesDir, 'profiles.yml');
+    let allProfiles;
+    try {
+        const raw = readFileSync(profilePath, { encoding: 'utf8' });
+        const rendered = renderProfilesYml(raw);
+        allProfiles = yaml.load(rendered) as Profiles;
+    } catch (e) {
+        throw new ParseError(
+            `Could not find a valid profiles.yml file at ${profilePath}:\n  ${e.message}`,
+        );
+    }
+    const profile = allProfiles[profileName];
+    if (!profile) {
+        throw new ParseError(
+            `Profile ${profileName} not found in ${profilePath}`,
+        );
+    }
+    const target = profile.outputs[targetName || profile.target];
```

Comment:
Should we throw an error if `target` is undefined ? e.g they typed the target name wrong ?